### PR TITLE
feat: migrate to Remix's built-in Tailwind plugin

### LINKED ISSUE #69: Switch to the newer TailwindCSS implementation

### PR DESCRIPTION
**Pull request by ricetech**
We've been using the Tailwind CLI to generate the Tailwind CSS file until now. This PR enables Remix's built-in Tailwind plugin and replaces the generated `tailwind.css` with the [proper file](https://tailwindcss.com/docs/guides/remix) that utilizes Tailwind's CSS directives. 

- Usages of Tailwind's CLI in `package.json` have been removed - this also means we no longer need `npm-run-all`. That has also been removed, and the `package.json` scripts have been simplified.
- The Tailwind config file is now written in TypeScript.
- If we upgrade to Remix@2.0.0, we can delete the manual enabling of the Tailwind plugin in `remix.config.js` since it is enabled by default in Remix@2.0.0.

**IMPORTANT: Before you pull, please delete `server/app/tailwind.css`.**

Fixes #69 (nice).

### CODE REVIEW

**Comment by ajayyy** [comment]
Why isn't `src/app/tailwind.css` deleted in the repo?

**Comment by ricetech** [comment]
> Why isn't `src/app/tailwind.css` deleted in the repo?

It was gitignored before because it was generated, but with the new implementation its purpose is completely different (it's where we import the Tailwind CSS files) and so it needs to be tracked now.

tl;dr it's a completely different file that happens to be at the same path

**Review by ajayyy — APPROVED** [review]
